recipes/libelf/all/conanfile.py

```diff
@@ -28,17 +29,20 @@ def _build_subfolder(self):
         return "build_subfolder"
 
     def config_options(self):
-        if self.settings.os != "Linux":
+        if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
-        if self.settings.os != "Linux":
-            if self.options.shared:
-                raise ConanInvalidConfiguration("libelf can not be built as shared library on non linux platforms")
+        if self.options.shared and self.settings.os not in ["Linux", "Windows"]:
+            raise ConanInvalidConfiguration("libelf can not be built as shared library on non linux or windows platforms")
```

Comment:
Is there any particular reason why libelf can not be build as a shared lib on MacOs?

Reply:
Because their autoconf file is not very portable.